prospero-common/src/main/java/org/wildfly/prospero/api/ProvisioningDefinition.java

```diff
@@ -104,23 +133,29 @@ private boolean channelsMissingManifest() {
         return channels.stream().anyMatch(c->c.getManifestRef() == null);
     }
 
-    private void setUpBuildEnv(List<Repository> additionalRepositories, Optional<Path> provisionConfigFile,
-                               Optional<ChannelManifestCoordinate> manifestRef, List<Channel> channels) throws IOException {
-        if (!provisionConfigFile.isPresent() && !manifestRef.isPresent()) {
-            if (!additionalRepositories.isEmpty()) {
-                this.channels = TemporaryRepositoriesHandler.addRepositories(channels, additionalRepositories);
+    // TODO: externalize
+    private static List<RemoteRepository> defaultChannelResolutionRepositories() {
+        return List.of(
+                new RemoteRepository.Builder("MRRC", null, "https://maven.repository.redhat.com/ga/").build(),
+                new RemoteRepository.Builder("central", null, "https://repo1.maven.org/maven2").build()
+        );
+    }
+
+    private static List<Repository> extractRepositoriesFromChannels(List<Channel> channels) {
+        return channels.stream().flatMap(c -> c.getRepositories().stream()).collect(Collectors.toList());
+    }
+
+    private void setKnownFeaturePackChannels(List<Repository> overrideRepositories,
```

Comment:
what happens if channelCoordinates is not empty?

Do you think this method could return List<Channel> and assign this.channels in the constructor?